### PR TITLE
[IMP] web_editor: add shortcut keys for link dialog

### PR DESCRIPTION
**Pull request by dhba-odoo**
Purpose:

1. Shortcut keys have been added to enhance the user experience for managing links
    within the link dialog box.
- alt+c: Save the link from the link dialog box.
- alt+j: Discard the link available inside the link dialog box.

2. Improve the user experience by transitioning the button-type widget from a checkbox
    to a selection dropdown.

3. Allow users to save a link or button by pressing Enter, provided that the URL is already
    inserted and the cursor is within the URL input box.

4. When creating a new link or editing an existing one, the current selection will collapse
    to the end of the link.

5. When editing the primary/secondary button, the size and style options will now be
    displayed by default.

task-3442763

### CODE REVIEW

**Comment by dmo-odoo** [comment]
This PR has a single commit but that commit seems to do multiple changes for multiple reasons. Please split the commit into multiple commits so that each commit encapsulates one specific requested change, with its own explanation of the reason why we want the change and how the change in the code solves the issue.

**Comment by dhba-odoo** [comment]
@dmo-odoo I have split the commit into multiple commits. You can take a look at them now.

File: addons/web_editor/static/src/xml/wysiwyg.xml [hunk 1]
```diff
@@ -66,7 +66,7 @@
                                 </div>
                             </div>
                             <div class="mb-3 row">
-                                <label class="col-form-label col-md-3">Style</label>
+                                <label class="col-form-label col-md-3">Layout</label>
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/xml/wysiwyg.xml:71]: This has nothing to do with the shortcuts and should be in another commit. I suggest to put it in the commit where Type is removed to Style since it is the reason why this other change had to be made as well.

File: addons/web_editor/static/src/xml/wysiwyg.xml [hunk 2]
```diff
@@ -31,7 +31,7 @@
                         <div t-attf-class="mb-3 row o_url_input#{state.isButton ? ' d-none' : ''}">
                             <label class="col-form-label col-md-3" for="o_link_dialog_url_input">URL or Email</label>
                             <div class="col-md-9">
-                                <input type="text" name="url" class="form-control" id="o_link_dialog_url_input" required="required" t-ref="inputUrl"/>
+                                <input type="text" name="url" class="form-control" id="o_link_dialog_url_input" required="required" t-ref="inputUrl" t-on-keydown="(ev) => ev.key === 'Enter' ? this.onSave(ev) : ''"/>
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/xml/wysiwyg.xml:34]: I think in owl you can name the property `t-on-keydown.prevent` to imply a `preventDefault` so you won't have to add it in the js above. Try it and let me know if it worked.

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js [hunk 3]
```diff
@@ -1471,8 +1472,8 @@ export class Wysiwyg extends Component {
                     this.odooEditor.historyUnpauseSteps();
                     this.odooEditor.historyStep();
                     const link = data.linkDialog.$link[0];
-                    setSelection(link, 0, link, link.childNodes.length, false);
-                    link.focus();
+                    const linkIndex = childNodeIndex(link);
+                    setSelection(link.parentElement, linkIndex+1, link.parentElement, linkIndex+1, false);
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1476]: I'm curious. What happens if you remove the two lines above? Isn't the selection at the place we want already?

File: addons/web_editor/static/tests/link_tests.js [hunk 4]
```diff
@@ -93,7 +93,7 @@ QUnit.module(
             await nextTick();
             assert.strictEqual(
                 editable.innerHTML,
-                `<p><a href="#" target="_blank">#</a><br></p>`
+                `<p><a href="#" target="_blank"><span data-o-link-zws="start" contenteditable="false">\u200B</span>#</a><span data-o-link-zws="after" contenteditable="false">\u200B</span><br></p>`
```
Comment (dmo-odoo) [inline addons/web_editor/static/tests/link_tests.js:96]: Try calling `editor.clean();` before like some tests below already does to avoid the zws getting in the way of the expected result.

**Review by dmo-odoo — COMMENTED** [review]
Thanks for splitting the changes, I have reviewed the PR.

File: addons/web_editor/static/src/xml/wysiwyg.xml (hunk 2 above)
Comment (dhba-odoo) [reply addons/web_editor/static/src/xml/wysiwyg.xml:34]: I want to prevent the `keydown` event for the Enter key only. However, using `t-on-keydown.prevent` prevents all `keydown` events, making it impossible to type anything in the URL input box.

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js (hunk 3 above)
Comment (dhba-odoo) [reply addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1476]: No, the selection isn't correct; without those lines, the selection will appear at the beginning of the editable area instead of after the link.

File: addons/web_editor/static/tests/link_tests.js (hunk 4 above)
Comment (dhba-odoo) [reply addons/web_editor/static/tests/link_tests.js]: I have updated the test cases as suggested.

File: addons/web_editor/static/src/xml/wysiwyg.xml (hunk 2 above)
Comment (dmo-odoo) [reply addons/web_editor/static/src/xml/wysiwyg.xml:34]: Oh, right, thanks for trying!

[the remark this answers is not in the document]
Comment (dhba-odoo) [reply addons/web_editor/static/src/xml/wysiwyg.xml]: Done!

**Comment by dmo-odoo** [comment]
@robodoo rebase-ff r+
